### PR TITLE
Revoke presign URLs

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
Added new function to file sdk to revoke all presign URLs 

## Description

method - `files.revokeAllPresignURLs()`
On sdk init along with creating container we will also add default access policy if it doesnt exist
all presign URLs will be generated based using this access policy identifier
While revoking URLs, we simply create access policy with new identifier

## How Has This Been Tested?
unit tests, and manual tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [x ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### CODE REVIEW

**Review by moritzraho — COMMENTED** [review]
### A general Question:
I wonder if we need a new uuid every time, can't we reuse the same ?
Have you tried the following?
```
policyName = 'aio-lib-files-policy', set policy always uses that id
Init:  do not get, set policy with same name => does this override the current one ? if not then we are good
Revoke: set policy with empty payload to remove then set again with same id
```
If that work it would be great as we wouldn't need to getAccessPolicy and could get rid of the additional request + cp/paste code of the  azure sdk

### Also see inline comments

**Comment by sandeep-paliwal** [comment]
@moritzraho regarding having the same poilcy id 'aio-lib-files-policy', this will not work, the entire revoke logic is based on generating signature based on a given Id, and then remove that id so that the presign URLs stop working. So a policy with empty payload is not enough, we need a policy with different Id each time we revoke.

File: lib/impl/AzureBlobFiles.js [hunk 1]
```diff
@@ -408,6 +416,105 @@ class AzureBlobFiles extends Files {
     }
   }
 
+  /**
+   * @memberof AzureBlobFiles
+   * @private
+   */
+  async _revokePresignURLs () {
```
[the remark this answers is not in the document]
Comment (sandeep-paliwal) [reply lib/impl/AzureBlobFiles.js:423]: Valid point @moritzraho , But there are quite a few helper methods within Azure impl file. I think this code refactoring can be done as a separate exercise possibly with Azure sdk update. WDYT?

[the remark this answers is not in the document]
Comment (moritzraho) [reply lib/impl/AzureBlobFiles.js:423]: okay :) just move them to the helper section of the file then

**Comment by moritzraho** [comment]
ok thanks for the details.
@sandeep-paliwal still tried and we saw that re-defining a removed policy gives access back to revoked presigned URLs. So the only way is to use a different id every time

**Review by moritzraho — APPROVED** [review]

**Pull request merged**